src/client/api/v1/send_message.c

```diff
@@ -107,7 +107,11 @@ int serialize_indexation(message_t* msg, byte_buf_t* buf) {
   byte_buf_free(hex_data);
   byte_buf_free(hex_index);
   if (json_string) {
+#ifdef _MSC_VER
```

Comment:
Is this change work on all MSVC compiler? if no, we should check the version number.
```
#if _MSC_VER >= 1920
```

Reply:
I tried Visual Studio 2017 (version 15.9) and Visual Studio 2019 (version 16.11) under different Platform Toolsets (v140 / v141 / v142).
I can't say anything for older versions.
For Visual Studio 2017 (version 15.9) _MSC_VER is defined as 1916.
I.e.:
```
#if defined (_MSC_VER)
#if (_MSC_VER >= 1916 )
    cJSON_free(json_string); 
#else
    free(json_string); // leads to an runtime-error
#endif
#else
    free(json_string);
#endif
```

Reply:
cJSON internal allocator uses default one which memory is released via `free()` and cJSON_free should work on different versions as well.
https://github.com/DaveGamble/cJSON/blob/master/cJSON.c#L165-L176
it's odd that free() gets a runtime error here. do you enable jemalloc or use different allocators in the project?

Reply:
Sorry, its my fault :-(  I have found this:

[Debug Versions of Heap Allocation Functions](https://docs.microsoft.com/en-us/visualstudio/debugger/debug-versions-of-heap-allocation-functions?view=vs-2019).

 I'am compiling the Example (data_message.c) with Debuginformations. But the Libraries i'am using were compiled without Debuginformation (Release). Now there are different Versions of Heap Allocation Functions (malloc and _malloc_dbg). This is the Reason of the Runtime-Error. When i use the Debug-Versions of the Libraries or i build a Release-Version of the Example (data_message.c) there is no Runtime-Error at all. There is no Reason to change the Code. Thanks for your patience.

Reply:
no worries, good to know the root cause :) 
please remove related code blocks and update this pull request, thanks.